bftengine/src/bftengine/BftCommandHandler.h

```diff
@@ -0,0 +1,26 @@
+#include <Replica.hpp>
+
+#pragma once
+
+namespace bftEngine {
+
+class BftCommandHandler : public IRequestsHandler {
```

Comment:
Is there a reason that it does not inherit from `ICommandHandler`?

Reply:
Yes, ICommandHandler has an additional method that is not necessary for this class.

Reply:
Are we talking about the same [class](https://github.com/vmware/concord-bft/blob/master/kvbc/include/KVBCInterfaces.h#L128)?

Reply:
Yes, ICommandsHandler is defined in kvbc layer :  kvbc/include/KVBCInterfaces.h

Reply:
@toly-kournik , I see, the kvbc is a higher level.
I think we have a problem in the hierarchy: 2 command handlers, where one inherits from `IRequestsHandler` and another from `ICommansHandler`. It is a bit misleading. Maybe we can move `ICommandsHandler` to the `bftEngine` namespace?

Reply:
To tell the truth I don't really understand the purpose of `ICommandsHandler`.
Probably it was designed to enhance `IRequestHandler` and be an API for KVBC, but currently it does nothing.
`setControlStateManager()` was added recently by @yontyon

Reply:
I was the one who recommended this:)
I do not understand the purpose of IRequestsHandler because nobody inherits from it directly.
I'd be happy to get rid of one of them.

Reply:
Since `IRequestHandler` is now used in bftEngine it should remain.
`ICommandsHandler` on the other hand is used by concord

Reply:
This is a bad design of interfaces, I don't see any reason to inherit from ICommandsHandler and to Dummy implement stuff

Reply:
The basic question about the design is if the new command handler is a commands handler?
If it is then it is a bad design to inherit from something else than ICommandsHandler.
Regarding, the dummy implementation, I think we might want to have a few base types for handlers, etc.

I understand that this is beyond the scope of the PR, the feature and obviously I do not ask to change it but we have to make a decision.
Otherwise, the inheritance model of the project will end up being ridiculously complex and unclear.

Reply:
Multiple inheritance implies bad design in my opinion, also over complex interfaces. If one wants to use an interface but needs to implement dummy methods, something is not correct.

Reply:
Let us take it offline:)